src/action_install.rs

```diff
@@ -81,85 +86,62 @@ fn show_install_summary(pacman_deps: &IndexSet<String>, aur_packages: &IndexMap<
+			&vec![review_dir],
+			rua_files::global_build_dir(dirs),
+			&CopyOptions::new(),
+		)
+		.expect("failed to copy reviewed dir to build dir");
+		rm_rf::force_remove_all(build_dir.join(".git")).expect("Failed to remove .git");
+		wrapped::build_directory(
+			&build_dir.to_str().expect("Non-UTF8 directory name"),
+			dirs,
+			offline,
+		);
+	}
+	for base in &actions.build {
+		check_tars_and_move(&base.pkgbase, dirs, &archive_whitelist);
+	}
+	// This relation between split_name and the archive file is not actually correct here.
+	// Instead, all archive files of some group will be bound to one split name only here.
+	// This is probably still good enough for install verification though --
+	// and we only use this relation for this purpose. Feel free to improve, if you want...
+	let mut files_to_install: Vec<(String, PathBuf)> = Vec::new();
+	for base in &actions.build {
+		let pkgbase = &base.pkgbase;
+		let checked_tars = rua_files::checked_tars_dir(dirs, &pkgbase);
+		let read_dir_iterator = fs::read_dir(checked_tars).unwrap_or_else(|e| {
+			panic!(
+				"Failed to read 'checked_tars' directory for {}, {}",
+				pkgbase, e
 			)
-			.expect("failed to copy reviewed dir to build dir");
-			rm_rf::force_remove_all(build_dir.join(".git")).expect("Failed to remove .git");
-			wrapped::build_directory(
-				&build_dir.to_str().expect("Non-UTF8 directory name"),
-				dirs,
-				offline,
-			);
-		}
-		for (pkgbase, _depth, _split) in &packages {
-			check_tars_and_move(pkgbase, dirs, &archive_whitelist);
-		}
-		// This relation between split_name and the archive file is not actually correct here.
-		// Instead, all archive files of some group will be bound to one split name only here.
-		// This is probably still good enough for install verification though --
-		// and we only use this relation for this purpose. Feel free to improve, if you want...
-		let mut files_to_install: Vec<(String, PathBuf)> = Vec::new();
-		for (pkgbase, _depth, split) in &packages {
-			let checked_tars = rua_files::checked_tars_dir(dirs, &pkgbase);
-			let read_dir_iterator = fs::read_dir(checked_tars).unwrap_or_else(|e| {
-				panic!(
-					"Failed to read 'checked_tars' directory for {}, {}",
-					pkgbase, e
-				)
-			});
-
-			for file in read_dir_iterator {
-				files_to_install.push((
-					split.to_string(),
-					file.expect("Failed to access checked_tars dir").path(),
-				));
-			}
+		});
+
+		for file in read_dir_iterator {
+			files_to_install.push((
+				pkgbase.to_string(),
+				file.expect("Failed to access checked_tars dir").path(),
+			));
 		}
-		pacman::ensure_aur_packages_installed(files_to_install, asdeps || depth > 0);
 	}
+
+	pacman::ensure_aur_packages_installed(files_to_install, true || as_deps); //TODO hande as_deps properly
 }
 
-pub fn check_tars_and_move(name: &str, dirs: &ProjectDirs, archive_whitelist: &IndexSet<&str>) {
+pub fn check_tars_and_move(name: &str, dirs: &ProjectDirs, archive_whitelist: &[&str]) {
```

Comment:
This part of the whole "aur-depends has been inserted into the
middle of the existing system. There's probably more that could be done
to reduce the code count futher, by removing thing's aur-depends can
already handle." point.

I don't actually see a reason for this variable to exist at all. If I read correctly, it's a set of all package names? Instead you can just get the package names for an individual base at a time. Which is a much smaller list.

Reply:
@Morganamilo yes, that's one possible improvement (not a big one though as IndexSet/HashSet is very fast on `.contains` method regardless).

An even better improvement would be to invoke `makepkg --packagelist` or `makepkg --printsrcinfo`. I think we should strive for the latter. But still, I see no benefits of changing an IndexSet to array here..

Reply:
Because hashmap like structures are not magically faster. They have overhead and will be slower unless the array is sufficiently large. And given that most packages are not split packages, I estimate the average size of this array to be 1.